File: video-series-query/src/main/java/org/jesperancinha/video/query/data/VideoSeries.java
package org.jesperancinha.video.query.data;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.Hibernate;
import org.jesperancinha.video.core.data.Genre;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.util.Objects;

import static jakarta.persistence.GenerationType.IDENTITY;

@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "VIDEO_SERIES")
public class VideoSeries {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    @Column
    private Long id;

    @Column
    private String name;

    @Column
    private Integer volumes;

    @Column
    private BigDecimal cashValue;

    @Column
    @Enumerated(EnumType.STRING)
    private Genre genre;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        VideoSeries that = (VideoSeries) o;

        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId(), getName(), getVolumes(), getCashValue(), getGenre());
    }
}

File: video-series-query/src/main/kotlin/org/jesperancinha/video/query/data/VideoSeries.kt
package org.jesperancinha.video.query.data

import jakarta.persistence.*
import org.jesperancinha.video.core.data.Genre
import java.math.BigDecimal
import java.util.*

@Entity
@Table(name = "VIDEO_SERIES")
data class VideoSeries(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column
    private val id: Long? = null,

    @Column val name: String,

    @Column val volumes: Int,

    @Column val cashValue: BigDecimal,

    @Column
    @Enumerated(EnumType.STRING) val genre: Genre,
)